### PR TITLE
Prepare Turbine for Publishing

### LINKED ISSUE #3: [Turbine Stage 3] Prepare Turbine-py for PyPI Publishing
Part of https://github.com/meroxa/turbine-project/issues/103

Ensure that each of the following files / directories are set up in the repo:
- Package license
- `pyproject.toml` 
- metadata files 
  -  This repo currently uses `setup.py`, the current preferences is for `setup.cfg`, the static version. [See more here](https://packaging.python.org/en/latest/tutorials/packaging-projects/#configuring-metadata)
- a `README` (may need to reach out to devrel to make sure we're consistent with other projects) (https://github.com/meroxa/turbine-project/issues/184)
- a `MANIFEST.in` 
  -  This may be handled in another task...

### PR DESCRIPTION
**Pull request by ericcheatham**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #3 

## Type of change

Please delete options that are not relevant.

- [ ] New feature
- [x] Bug fix
- [ ] Refactor
- [x] Documentation

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Unit Tests
- [x] Manual Tests
- [x] Deployed to staging

# Additional references

*Any additional links (if appropriate)*

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,129 @@
-/build
```
Comment (ericcheatham) [inline .gitignore:1]: Copied the python suggested git ignore from github. It doesn't play around.

File: src/turbine/runner/runner.py [hunk 2]
```diff
@@ -35,10 +36,12 @@ def clean_temp_directory(tmp_dir):
         shutil.rmtree(tmp_dir, ignore_errors=True)
 
     async def run_app_platform(self, image_name):
+        parsed_url = urlparse(os.environ.get("MEROXA_API_URL"))
+
         environment = PlatformRuntime(
             client_options=ClientOptions(
                 auth=os.environ.get("MEROXA_ACCESS_TOKEN"),
-                url=os.environ.get("MEROXA_API_URL"),
```
Comment (ericcheatham) [inline src/turbine/runner/runner.py:41]: The environment variable of the same name is used in the cli. The library we use to make REST requests wants the base URL. This update parses the URL into its components: scheme, netloc, and path. The library only wants the scheme and the netloc.

File: src/turbine/runtime/platform.py [hunk 3]
```diff
@@ -214,15 +214,19 @@ async def process(
         self, records: Records, fn: t.Callable[[t.List[Record]], t.List[Record]]
     ) -> Records:
 
+        pipeline_id = PipelineIdentifiers(
```
Comment (ericcheatham) [inline src/turbine/runtime/platform.py:217]: This was causing issues from time to time with NoneType errors. I think it was caused by the getters/setters accessing a parameter before those parameters actually had any sort of physical value.

File: setup.cfg [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+[options]
+packages = find:
+package_dir = = src
+include_package_data = True
+python_requires = >=3.7
+
+install_requires =
+    aiohttp>=3.8
+    python-dotenv==0.20.0
```
Comment (janelletavares) [inline setup.cfg:28]: i don't think this one is required anymore

**Review by janelletavares — APPROVED** [review]
just one observation

**Pull request merged**